tests/http/abnfRules/uriRulesTester.cpp

```diff
@@ -0,0 +1,742 @@
+{
+  // TODO if needed
+}
+
+/**
+ * relative-ref = relative-part [ "?" query ] [ "#" fragment ]
+ */
+TEST(UriAbnfTest, RelativeRef)
+{
+  // TODO if needed
+}
+
+/**
+ * relative-part = "//" authority path-abempty
+ *               / path-absolute
+ *               / path-noscheme
+ *               / path-empty
+ */
+TEST(UriAbnfTest, RelativePart)
+{
+  // TODO if needed
+}
+
+/**
+ * scheme = ALPHA *( ALPHA / DIGIT / "+" / "-" / "." )
+ */
+TEST(UriAbnfTest, Scheme)
+{
+  SequenceRule* sequenze = schemeRule();
+  EXPECT_TRUE(runParser("a1234", *sequenze));
+  EXPECT_TRUE(runParser("a", *sequenze));
+  EXPECT_TRUE(runParser("a+", *sequenze));
+  EXPECT_TRUE(runParser("a-", *sequenze));
+  EXPECT_TRUE(runParser("a.", *sequenze));
+
+  // Invalid
+  EXPECT_FALSE(runParser("1a1234", *sequenze));
+}
+
+/**
+ * authority = [ userinfo "@" ] host [ ":" port ]
+ */
+TEST(UriAbnfTest, Authority)
+{
+  SequenceRule* sequence = authorityRule();
+  // --- host only ---
+  EXPECT_TRUE(runParser("example.com", *sequence));
+  EXPECT_TRUE(runParser("localhost", *sequence));
+  EXPECT_TRUE(runParser("192.168.0.1", *sequence));
+  // EXPECT_TRUE(runParser("[2001:db8::1]", *sequence)); // Todo Ipv6
+  EXPECT_TRUE(runParser("sub.domain.co.uk", *sequence));
+
+  // --- host + port ---
+  EXPECT_TRUE(runParser("example.com:80", *sequence));
+  EXPECT_TRUE(runParser("localhost:8080", *sequence));
+  EXPECT_TRUE(runParser("192.168.0.1:65535", *sequence));
+  // EXPECT_TRUE(runParser("[2001:db8::1]:443", *sequence)); // Todo Ipv6
+
+  // --- userinfo + host ---
+  EXPECT_TRUE(runParser("@example.com", *sequence));
+  EXPECT_TRUE(runParser("user@example.com", *sequence));
+  EXPECT_TRUE(runParser("user:pass@example.com", *sequence));
+  EXPECT_TRUE(runParser("admin@127.0.0.1", *sequence));
+  // EXPECT_TRUE(runParser("root@[::1]", *sequence));  // Todo Ipv6
+
+  // --- userinfo + host + port ---
+  EXPECT_TRUE(runParser("user@example.com:80", *sequence));
+  EXPECT_TRUE(runParser("user:pw@host.net:8080", *sequence));
+  // EXPECT_TRUE(runParser("me@[2001:db8::2]:443", *sequence));  // Todo Ipv6
+
+  // Other
+  EXPECT_TRUE(runParser("user@", *sequence));
+  EXPECT_TRUE(runParser("user@:80", *sequence));
+  EXPECT_TRUE(runParser("user@host:99999", *sequence));
+
+  // Invalid
+  EXPECT_FALSE(runParser("example.com: ", *sequence));
+  EXPECT_FALSE(runParser("user@host:a", *sequence));
+  EXPECT_FALSE(runParser("user@host:12:34", *sequence));
+}
+
+/**
+ * userinfo = *( unreserved / pct-encoded / sub-delims / ":" )
+ */
+TEST(UriAbnfTest, UserInfo)
+{
+  RepetitionRule* rep = userinfoRule();
+  EXPECT_TRUE(runParser("a1234", *rep));
+  EXPECT_TRUE(runParser("a:", *rep));
+}
+
+/**
+ * host = IP-literal / IPv4address / reg-name
+ */
+TEST(UriAbnfTest, Host)
+{
+  // TODO if needed
+}
+
+/**
+ * port = *DIGIT
+ */
+TEST(UriAbnfTest, Port)
+{
+  AlternativRule* alter = hostRule();
+  EXPECT_FALSE(runParser("example.com:abc", *alter));
+  delete alter;
+}
```

Comment:
Should this test be rather like this?
```suggestion
/**
 * port = *DIGIT
 */
TEST(UriAbnfTest, Port)
{
  RepetitionRule* rep = portRule();
  EXPECT_FALSE(runParser("abc", *rep));
  EXPECT_TRUE(runParser("8080", *rep));
  delete rep;
}
```